### ISSUE TITLE
Question about commercial use and AGPL interpretation

### ISSUE THREAD

**Issue by ohgyeongsu** [ultralytics/ultralytics#21936]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

We trained a YOLOv8 model using the Ultralytics codebase with our own dataset. After training, we obtained a .pt file. Our intended usage scenario is as follows:

1. Convert the resulting .pt file into a DLL and integrate it into our equipment.
2. Within our equipment, use this DLL to perform object detection inspection on images.
3. Deliver our equipment to another company. The other company uses the equipment, but they cannot access or view the .pt file itself.

Under the AGPL license, would this scenario be considered internal use or distribution?

Additionally, if we are required to release code, do we need to disclose:

(1) Preprocessing steps,

(2) The part where we import ultralytics,

(3) The training scripts/process,

(4) Inference
or some other specific components?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ohgyeongsu, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you started while an Ultralytics engineer reviews your question and will assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

For licensing/commercial use questions like yours, please share additional context so our team can provide accurate guidance:
-Exact ultralytics package version and PyTorch version
-High-level description of deployment architecture (e.g., where inference runs, how the DLL is called, network access, whether any APIs are exposed)
-Whether any modifications were made to the Ultralytics source code
-A brief summary of which components are redistributed to third parties vs. kept internal

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

👍 Once you share the additional details above, our team can review and follow up with next steps.

**Comment by glenn-jocher** [comment]
Thanks for the clear scenario—shipping your equipment with a DLL that embeds a YOLOv8‑derived `.pt` to another company is distribution, not internal use, so AGPL‑3.0 obligations apply regardless of the closed network or DLL wrapping. To rely on AGPL you must open‑source the complete corresponding source of the larger work under AGPL‑3.0, including your preprocessing, code that imports/integrates Ultralytics (e.g., the DLL wrapper), training and inference code, configs/environment, and the fine‑tuned weights; if applicable, also provide installation information for recipients—see the [AGPL‑3.0 terms](https://www.ultralytics.com/legal/agpl-3-0-software-license) and our guide on [open‑sourcing your YOLO project](https://docs.ultralytics.com/help/contributing/). If you need to keep this proprietary, please use the [Ultralytics Enterprise License](https://www.ultralytics.com/license); we can’t offer legal advice, so consult your counsel for a final determination.
